Singular Documentation Updates (#455)
rudderlabs/rudderstack-docs: destinations/analytics/singular.mdx

- **API Key**: Enter your Singular API key here. This is a mandatory field. 
- **Secret**: Enter your Singular secret. This is a mandatory field and is required for the device mode integrations with the RudderStack iOS and Android SDKs.

<div class="infoBlock">

For more information on obtaining the Singular API key and secret, refer to the <a href="#faq">FAQ</a> section below.
</div>

<div class="successBlock">

When sending events via the device mode, RudderStack also lets you specify which events should be discarded or allowed to flow through. For more information, refer to the <a href="https://www.rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/event-filtering/"  target="_blank">Client-side Event Filtering</a>.
</div>

## Adding device mode integration

To add Singular to your application, follow the steps below depending on your platform of integration.

<Tabs>
  <TabList>
    <Tab>iOS</Tab>
    <Tab>Android</Tab>
    <Tab>React Native</Tab>
    <Tab>Cordova</Tab>
  </TabList>
    <TabPanels>
      <TabPanel>
        To add Singular to your iOS app, follow these steps:

> <ul>

Reviewer: ```suggestion
<ol>
```
So that the steps are numbered

```objectivec
RSConfigBuilder *configBuilder = [[RSConfigBuilder alloc] init];
[configBuilder withDataPlaneUrl:<DATA_PLANE_URL>];
[configBuilder withFactory:[RudderSingularFactory instance]];
RSClient *rudderClient = [RSClient getInstance:<WRITE_KEY> config:[configBuilder build]];
```
</span>
</li>

<div class="infoBlock">

For more information, refer to the <a href="https://support.singular.net/hc/en-us/articles/360037950591-iOS-SDK-Basic-Integration?navigation_side_bar=true" target="_blank">Singular iOS documentation</a>.
</div>

> </ul>

Reviewer: ```suggestion
</ol>
```

same can be followed for all steps in different tabs

<span>

```typescript
import rudderClient from "@rudderstack/rudder-sdk-react-native"
import singular from "@rudderstack/rudder-integration-singular-react-native"
const config = {
	dataPlaneUrl: <DATA_PLANE_URL>,
	trackAppLifecycleEvents: true,
	withFactories: [singular],
}
rudderClient.setup(<WRITE_KEY>, config)
```
</span>
</li>
</ul>
      </TabPanel>
      <TabPanel>
        To add Singular to your Cordova app, follow these steps:
<ul>
<li>Navigate to the root folder of your application and run the following command:
<span>

```bash
cordova plugin add rudder-integration-singular-cordova
```
</span>
</li>
<li>Then, add the following code in the <code class="inline-code">onDeviceReady()</code> function of your app's home page to initialize the SDK, as shown:
<span>

```bash
RudderClient.initialize(<WRITE_KEY>, {
  dataPlaneUrl: <DATA_PLANE_URL>,
  factories: [RudderSingularFactory]
})
```
</span>
</li>
<div class="infoBlock">

Make sure you use the <code class="inline-code">await</code> keyword with the <code class="inline-code">initialize</code> call.
</div>
</ul>
      </TabPanel>
    </TabPanels>
</Tabs>

## Implementing SKAdNetwork (SKAN) support

> To give Singular the control to your SKAdNetwork integration, add the following code before the initialization of the iOS SDK:

Reviewer: ```suggestion
Add the following code before the initialization of the iOS SDK to give the control to Singular for your SKAdNetwork integration:
```

<span>

```bash
npm install @rudderstack/rudder-integration-singular-react-native
// OR //
yarn add @rudderstack/rudder-integration-singular-react-native
```
</span>
</li>
<li>Then, import the above module and add it to your SDK initialization, as shown:
<span>

```typescript
import rudderClient from "@rudderstack/rudder-sdk-react-native"
import singular from "@rudderstack/rudder-integration-singular-react-native"
const config = {
	dataPlaneUrl: <DATA_PLANE_URL>,
	trackAppLifecycleEvents: true,
	withFactories: [singular],
}
rudderClient.setup(<WRITE_KEY>, config)
```
</span>
</li>
</ul>
      </TabPanel>
      <TabPanel>
        To add Singular to your Cordova app, follow these steps:
<ul>
<li>Navigate to the root folder of your application and run the following command:
<span>

```bash
cordova plugin add rudder-integration-singular-cordova
```
</span>
</li>
<li>Add the platforms you want to target for your app:
<span>

```bash
cordova platform add ios
cordova platform add android
```
</span>
</li>
<li>Run the following command to build the app for all the platforms:
<span>

```bash
cordova build
```
</span>

> </li>

Reviewer 2: We can drop these lines

A sample `identify` call for both the Android and iOS SDKs is shown below:

<Tabs>
  <TabList>
    <Tab>iOS</Tab>
    <Tab>Android</Tab>
  </TabList>
    <TabPanels>
      <TabPanel>
<span>

```objectivec
[[RSClient getInstance] identify:@"1hKOmRA4el9Zt1WSfVJIVo4GRlm"];
```
</span>
      </TabPanel>
      <TabPanel>
<span>

```kotlin
RudderClient.getInstance()?.identify("1hKOmRA4el9Zt1WSfVJIVo4GRlm")
```
</span>
      </TabPanel>
    </TabPanels>
</Tabs>

## Track

The [`track`](https://www.rudderstack.com/docs/rudderstack-api/api-specification/rudderstack-spec/track/) call lets you capture any user actions and the properties associated with them. Each user action is considered to be an event.

### Tracking custom events

A custom `track` call lets you track custom events as they occur in your apps. RudderStack sends these calls to Singular where they are processed as custom post-install events and are made available in the relevant reports.

A sample custom `track` call for both the Android and iOS SDKs is shown below:

<Tabs>
  <TabList>
    <Tab>iOS</Tab>
    <Tab>Android</Tab>
  </TabList>
    <TabPanels>
      <TabPanel>
<span>

```objectivec

> "[[RSClient getInstance] track:@""Product Reviewed"" properties:@{
>         @""product_id"" : @""345676543"",
>         @""review_id"" : @""123454387""
>     }];"

Reviewer 3: ```suggestion
[[RSClient getInstance] track:@"Product Reviewed" properties:@{
        @"product_id" : @"345676543",
        @"review_id" : @"123454387"
    }];
```

```
</span>
      </TabPanel>
      <TabPanel>
<span>

```kotlin
RudderClient.getInstance()?.track("Product Reviewed",
    RudderProperty()
        .putValue("product_id", "345676543")
        .putValue("review_id", "123454387")
)
```
</span>
      </TabPanel>
    </TabPanels>
</Tabs>

### Tracking revenue

> Singular supports tracking revenue events. It receives revenue tracking whenever an event containing the `revenue` property is sent(including a zero value). Optionally, you can also pass the `currency` field as an [ISO code](https://www.iso.org/iso-4217-currency-codes.html). 

Reviewer: 'It receives revenue tracking' seems bit unclear.

Does it mean 'It tracks the revenue whenever an event containing the `revenue` property is sent(including a zero value).'?

```
</span>
      </TabPanel>
    </TabPanels>
</Tabs>

## Screen

The [`screen`](https://www.rudderstack.com/docs/rudderstack-api/api-specification/rudderstack-spec/screen/) method allows you to record whenever a user sees the mobile screen, along with any associated optional properties. This call is similar to the `page` call for the web applications but exclusive to your mobile device.

A sample `screen` call for both the Android and iOS SDKs is shown below:

<Tabs>
  <TabList>
    <Tab>iOS</Tab>
    <Tab>Android</Tab>
  </TabList>
    <TabPanels>
      <TabPanel>
<span>

```objectivec
[[RSClient sharedInstance] screen:@"Home" properties:@{
    @"category" : @"launcher"
}];
```
</span>
      </TabPanel>
      <TabPanel>
<span>

```kotlin
RudderClient.getInstance()?.screen("Home",
    RudderProperty()
        .putValue("category", "launcher")
)
```
</span>
      </TabPanel>
    </TabPanels>
</Tabs>

In the above snippet, RudderStack captures all the information related to the viewed screen, along with any additional info about the screen.

<div class="infoBlock">

RudderStack sends the <code class="inline-code">screen</code> event to Singular as a custom event.
</div>

## Reset

> The `reset` method resets the current user's identity and creates a new anonymous session. It should be called when the users log out.

Reviewer: ```suggestion
The `reset` method resets the current user's identity and creates a new anonymous session. It should be called when a user logs out.
```
